dis_003_volcano_reports/contents/src/__init__.py

```diff
@@ -30,184 +38,317 @@
-###
+    return(num_dropped)
 
 def genUID(lat,lon,dt):
+    '''Generate unique id using latitude, longitude and date information
+    INPUT   lat: latitude of the earthquake (float)
+            lon: longitude of the earthquake (float)
+            dt: date for which we want to generate id (string)
+    '''
     return '{}_{}_{}'.format(lat,lon,dt)
 
-def processData(existing_ids):
-    """
-    Inputs: FTP SOURCE_URL and filename where data is stored, existing_ids not to duplicate
-    Actions: Retrives data, dedupes and formats it, and adds to Carto table
-    Output: Number of new rows added
-    """
+def processData(url, existing_ids):
+    '''
+    Fetch, process and upload new data
+    INPUT   url: url where you can find the source data (string)
+            existing_ids: list of date IDs that we already have in our Carto table (list of strings)
+    RETURN  num_new: number of rows of new data sent to Carto table (integer)
+    '''
+    # create an empty list to store new data
     new_data = []
+    # create an empty list to store unique ids of new data we will be sending to Carto table
     new_ids = []
-
-    res = req.get(SOURCE_URL)
+    # pull data from the url
+    res = req.get(url)
+    # break down the content retrieved from the url
     xml = lxml.etree.fromstring(res.content)
+    # convert xml into Python dictionary structure (json in this case)
     json = xml2json.data(xml)
+    # get each volcano report from the item feature in json
     items = json['channel']['item']
-
+    # loop through to process each volcano report
     for item in items:
+        # get the volcano name and country name from the title 
         title = item['title'].split(')')[0].split('(')
+        # remove leading and trailing whitespaces from title
         place_info = [place.strip() for place in title]
+        # get the volcano name from first index of place_info
         volcano_name = place_info[0]
+        # get the country name from second index of place_info
         country_name = place_info[1]
-
+        # get the coordinates and split it to separate out latitude and longitude
         coords = item['{http://www.georss.org/georss}point'].split(' ')
-        dt = parser.parse(item['pubDate'], fuzzy=True).strftime(DATETIME_FORMAT)
-
+        # get the longitude from first index of coords
         lat = coords[0]
+        # get the latitude from second index of coords
         lon = coords[1]
+        # create a geojson from the coordinates
         geom = {
             'type':'Point',
             'coordinates':[lon,lat]
         }
-
+        # get the date from pubDate feature and format according to DATETIME_FORMAT
+        dt = parser.parse(item['pubDate'], fuzzy=True).strftime(DATETIME_FORMAT)
+        # get the description and source of the volcano report from description feature
         info = item['description'].split('Source:')
+        # if the word 'Source:' don't exist in this item, i.e. if there are more than one source
```

Comment:
```suggestion
        # if the word 'Source:' doesn't exist in this item, it may be because there is more than one source (labeled as 'Sources')
```